docs/references/architecture/adr-112-txi.md

```diff
@@ -0,0 +1,84 @@
+# ADR 112: Modular transaction hashing
+
+## Changelog
+
+- 2024-02-05: First version (@melekes)
+
+## Status
+
+TBD
+
+## Context
+
+Transaction hashing in CometBFT is currently implemented using `tmhash`
+package, which itself relies on `sha256` to calculate a transaction's hash.
+
+The hash is then used by the built-in indexer (to index this
+transaction) and by the RPC `tx` and `tx_search` endpoints, which allow users
+to search for a transaction using a hash.
+
+`tmhash` is also used to calculate various hashes in CometBFT (e.g.,
+evidence, consensus params, commit, header, partset header).
```

Comment:
Not true for commit, header and partset header. 

They use the `merkle` package which, it is true, uses `sha256`.

Reply:
> Not true for commit, header and partset header.
> They use the merkle package which, it is true, uses sha256.

https://github.com/cometbft/cometbft/blob/d02b25e0fd278bc7e70142190ae5b3b8e722a8df/crypto/merkle/hash.go

are you sure?

Reply:
True, in the verification side, sorry.